mvpa2/generators/resampling.py

```diff
@@ -76,31 +76,38 @@ def __init__(self,
           Name of the selection marker attribute in the output dataset that is
           created if the balanced selection is not applied to the output dataset
           (see ``apply_selection`` argument).
+        rng : int or RandomState, optional
+          Integer to seed a new RandomState upon each call, or instance of the
+          numpy.random.RandomState to be reused across calls. If None, the
+          numpy.random singleton would be used
         """
         Node.__init__(self, space=space, **kwargs)
         self._amount = amount
         self._attr = attr
         self.count = count
         self._limit = limit
-        self._limit_filter = None
         self._include_offlimit = include_offlimit
         self._apply_selection = apply_selection
+        self._rng = rng
 
-
-    def _call(self, ds):
+    def _get_call_kwargs(self, ds):
+        attr, collection = ds.get_attr(self._attr)
+        # _call might need to operate on the dedicated instantiated rng
+        # e.g. if seed int is provided
+        return {
+            'limit_filter': get_limit_filter(self._limit, collection),
+            'rng': get_rng(self._rng)
+        }
+
+    def _call(self, ds, limit_filter=None, rng=None):
```

Comment:
Well, it is more just to make them keyword arguments... There is no way to enforce them being provided though in python 2 afaik... But they always should be provided unless someone provides wrong `_call_kwargs`... So overall I wouldn't sweat about assuring then not being None for this one ... Tests pass ;-)